### ISSUE TITLE
Display graphics and pretty printed formulae with `dap-ui-repl`.

### ISSUE THREAD

**Issue by hongyi-zhao** [emacs-lsp/dap-mode#521]
Currently, I'm using the [comint-mime](https://github.com/astoff/comint-mime) package with the following configuration:

```emacs-lisp
(use-package comint-mime
  :straight (:host github :repo "astoff/comint-mime"
             :files (:defaults "*.py" "*.sh"))
  :hook ((shell-mode inferior-python-mode) . comint-mime-setup))
```
With the help of the above package, Emacs can display graphics and pretty printed formulae as follows, just as I've discussed [here](https://github.com/astoff/comint-mime/issues/2#issuecomment-922462074):

![image](https://user-images.githubusercontent.com/11155854/136348787-15939159-c057-477b-a667-692bf784bcfc.png)

But I want to know if we can get the similar feature in `dap-ui-repl`.

Regards,
HZ

**Comment by yyoncho** [comment]
ATM this is not possible using dap-ui-repl. Somebody has to investigate and implement it if possible.

**Comment by amirdib** [comment]
Hi, 
Is there a way to, at least, use ipython shell in dap-ui-repl ?

**Comment by yyoncho** [comment]
@amirdib in general, you may run an external python process that supports graphics and connect using dap-mode.

**Comment by hongyi-zhao** [comment]
@yyoncho Does the [comint-mime](https://github.com/astoff/comint-mime) based example I showed above fall into the situation you said?

**Comment by amirdib** [comment]
> @amirdib in general, you may run an external python process that supports graphics and connect using dap-mode.

Can you point to the relevant documentation entry ? I can't find what you're referring to. If so, that would solve the issue.

**Comment by yyoncho** [comment]
reffer to that, and afaics there is attach template registered in dap-python: https://github.com/microsoft/debugpy#attaching-to-a-running-process-by-id

**Comment by hongyi-zhao** [comment]
Thank you for your linked documentation. But if I start ipython, the PID should be generated/assigned randomly completely. So, I still can't figure out how to use the example given there.

**Comment by amirdib** [comment]
> reffer to that, and afaics there is attach template registered in dap-python: https://github.com/microsoft/debugpy#attaching-to-a-running-process-by-id

Thanks. Indeed, there is an attach option: https://code.visualstudio.com/docs/python/debugging

I think PID is not needed. Next step will be to try on a minimal example @hongyi-zhao

**Comment by hongyi-zhao** [comment]
@amirdib If you have figured out one, hope you can follow up and post here.

**Comment by amirdib** [comment]
> Thank you for your linked documentation. But if I start a process, say `ipython`, the `PID` should be generated/assigned randomly completely. So, I still can't figure out how to use the example given there.

You can find the PID of a process by name with pgrep. So, `python3 -m debugpy --listen localhost:5882 --pid $(pgrep -f main.py)` will inject in debugger in main.py

**Comment by hongyi-zhao** [comment]
Run this from within Emacs?

**Comment by amirdib** [comment]
> Still not so clear. Run this from within Emacs? Could you please describe the specific steps?
I don't think that the solution proposed by @yyoncho correspond to what we intend to do. We want our debug shell to be a comint buffer that can display graphics. For that, would likely require to investigate and rewrite the `dap-ui-repl` function
